x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/AbstractNativeProcess.java

```diff
@@ -223,7 +223,7 @@ public void close() throws IOException {
      * Implementations can override this if they need to perform extra processing
      * immediately after the native process's input stream is closed.
      */
-    protected void afterProcessInStreamClose() {
+    protected void afterProcessInStreamClose() throws TimeoutException {
```

Comment:
maybe it would be clearer to say "closed" rather than "close"?

Reply:
Let's leave this minor naming concern for a followup. Dave K is off for the rest of the year and if this change is merged today it will be in time for 8.12.0. I'll merge it now.